lib/ui/recommendation.js

```diff
@@ -123,21 +123,25 @@ Recommendation.prototype = {
     this.events.publish('recommendation-mousemove', evt);
   },
   show: function(data) {
+    let payload = data.payload;
+    let ResultClass = data.resultClass;
+
     if (this.timeout) {
       this.win.clearTimeout(this.timeout);
       this.timeout = null;
     }
 
     // Don't re-render the recommendation if the content hasn't changed since
     // the previous recommendation.
-    if (this.prev && this.el && this.el.firstChild && data.result.url === this.prev.result.url) {
+    if (this.prev && this.el && this.el.firstChild &&
+        payload.result.url === this.prev.result.url) {
       this.el.collapsed = false;
       this.events.publish('recommendation-shown', this);
       return;
     }
 
-    this.prev = data;
-    const recommendation = this.row = new this.RecommendationRow(data, {
+    this.prev = payload;
+    const recommendation = this.row = new ResultClass(payload, {
```

Comment:
Why do we need to do an `if (recommendation)` check at line 150 below? The constructor should always return an object, which is truthy